lib/src/collection/treeset.dart

```diff
@@ -929,17 +938,22 @@ class _AvlTreeIterator<V> implements BidirectionalIterator<V> {
   final bool reversed;
   final AvlTreeSet<V> tree;
   final int _modCountGuard;
-  final V anchorObject;
+  final V? anchorObject;
   final bool inclusive;
 
-  _IteratorMove _moveNext;
-  _IteratorMove _movePrevious;
+  late _IteratorMove _moveNext;
+  late _IteratorMove _movePrevious;
 
-  int state;
-  _TreeNode<V> _current;
+  late int state;
+  _TreeNode<V>? _current;
 
   @override
-  V get current => (state != WALK || _current == null) ? null : _current.object;
+  V get current {
+    if (state == WALK && _current != null) {
+      return _current!.object;
+    }
+    throw StateError('No current element');
```

Comment:
It's possible to make this change in a less breaking way; e.g. v2/sdk/lib/internal/iterable.dart:

```
    // Before NNBD, this returned null when iteration was complete. In order to
    // avoid a hard breaking change, we return "null as E" in that case so that
    // if strong checking is not enabled or E is nullable, the existing
    // behavior is preserved.
    if (_remaining < 0) return null as E;
    return _iterator.current;
  }
```

Reply:
Done.